lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -460,12 +461,23 @@ protected function _getCreateTableSQL($tableName, array $columns, array $options
     }
 
     /**
+     * Tells whether a field type supports declaration of a default value.
+     *
+     * MySQL (as of 5.7.19) does not support default values for Blob and Text
+     * columns while MariaDB 10.2.1 does.
+     */
+    protected function isDefaultValueSupportedForType(Type $field) : bool
+    {
+        return !($field instanceof TextType || $field instanceof BlobType);
```

Comment:
Rather invert the condition using `&&` and without parentheses.

Reply:
Are you sure on this ? :) 
tip: only possible: true && false, false && true, always gives false

Reply:
@belgattitude I guess it should be `!($field instanceof TextType) && !($field instanceof BlobType)` according to De Morgan: https://thepugautomatic.com/2012/09/de-morgans-laws-in-programming/ 
But I prefer your solution because it's a bit faster: `&&` always calculate both parts :)

Reply:
Hey @vicdeo, thanks. I guess @majkl578 meant inversing that way.  Just replied between 2 calls, and forgot to re read.

Reply:
.... my bad 😉I'll let out that way for now

Reply:
```php
return ! $field instanceof TextType && ! $field instanceof BlobType;
```

> But I prefer your solution because it's a bit faster: && always calculates both parts and || doesn't :)

Source? Afaik they're both compiled the same way: https://github.com/php/php-src/blob/php-7.2.0RC3/Zend/zend_compile.c#L7119
Anyway, this would have absolutely negligible impact, not even worth having this discussion. ^^

Reply:
Out of topic, but I can't ignore that mistake. @VicDeo said:

> `&&` always calculates both parts and `||` doesn't :)

This is incorrect. As demonstrated by those commands both `&&` and `||` have cases were they do not evaluate the second part:

 ```sh
$ php -r 'var_dump(true || thisNonExistingFunctionWillNeverBeCalled());'
Command line code:1:
bool(true)

$ php -r 'var_dump(false && thisNonExistingFunctionWillNeverBeCalled());'
Command line code:1:
bool(false)
```

Reply:
Done ! thanks for explanation